crates/rpc/src/dto/primitives.rs

```diff
@@ -0,0 +1,240 @@
+use crate::dto::serialize;
+
+use super::serialize::SerializeForVersion;
+
+pub struct SyncStatus<'a>(pub &'a crate::v02::types::syncing::Status);
+
+pub struct Felt<'a>(pub &'a pathfinder_crypto::Felt);
+pub struct BlockHash<'a>(pub &'a pathfinder_common::BlockHash);
+pub struct ChainId<'a>(pub &'a pathfinder_common::ChainId);
+pub struct BlockNumber(pub pathfinder_common::BlockNumber);
+
+mod hex_str {
+    use std::borrow::Cow;
+
+    const LUT: [u8; 16] = *b"0123456789abcdef";
+
+    pub fn bytes_to_hex_str_stripped(data: &[u8]) -> Cow<'static, str> {
+        // Skip empty bytes
+        let zero_count = data.iter().take_while(|b| **b == 0).count();
+        let data = &data[zero_count..];
+
+        if data.is_empty() {
+            return Cow::from("0x0");
+        }
+
+        // Is the most significant nibble zero? We need to skip it if it is.
+        // Index is safe since we just checked that it is non-empty.
+        let zero_nibble = (0xF0 & data[0]) == 0;
+        to_hex_str(dbg!(data), dbg!(zero_nibble)).into()
+    }
+
+    pub fn bytes_to_hex_str_full(data: &[u8]) -> String {
+        to_hex_str(data, false)
+    }
```

Comment:
Where do we use the full representation?

Reply:
For Ethereum address: https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L1322-L1328

Reply:
But not used in this PR